### PR TITLE
GEOPY-1270

### PR DESCRIPTION
**Pull request by domfournier**

### CODE REVIEW

File: docs/source/methodology.rst [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            Uses the `refine_tree_from_triangulation <refine_triangulation>`_ method.
+
+- **Levels**:
+    List of integers defining the number of cells requested at each octree level.
+
+    .. math::
+        [1^{st}, 2^{nd}, 3^{rd}, ...]
+
+- [Optional] **Define as horizon**:
+    If checked, the object will be used to define a horizon. The vertices of the object are used to first
```
Comment (benk-mira) [inline docs/source/methodology.rst:167]: This must only be for surface?  If so mention it here

File: docs/source/methodology.rst [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    create a Delaunay surface, which is then used to refine the mesh as layers of cells below the surface.
+    Uses the `refine_tree_from_surface <refine_surface>`_ method.
+
+    - **Maximum distance**:
```
Comment (benk-mira) [inline docs/source/methodology.rst:171]: I think this was supposed to be at the same indent as Object and Levels.  Might I suggest shuffling the order too?:
image

File: docs/source/methodology.rst [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+In the example below, the mesh is refined from the vertices of Points object. The parameters are as follows:
+
+
+.. image:: images/octree_radial.png
+  :width: 800
+  :alt: radial
+
+
+This results in a mesh that has 4 concentric shells of cells at 25 m, followed by 4 cells at 50 m around each vertex.
+Note that the refinement is continuous only at the 2th octree level (50 m) where the refinements overlap.
```
Comment (benk-mira) [inline docs/source/methodology.rst:197]:
```suggestion
Note that the refinement is continuous only at the 2nd octree level (50 m) where the refinements overlap.
```

File: docs/source/methodology.rst (hunk 1 above)
Comment (benk-mira) [reply docs/source/methodology.rst:167]: Is this equivalent to the 'Refine by layers' method covered below?

**Review by benk-mira — CHANGES_REQUESTED** [review]
Lookin good!  See suggestions/comments.

File: docs/source/usage.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+From command line
+-----------------
+
+The application can also be run from the command line.  This is useful for more advanced users that may want to automate
+the mesh creation process, or re-run an existing mesh with different parameters.  To run the application from the command line,
+
+``python -m octree_creation_app.driver input_file.json``
+
+where ``input_file.json`` is the path to the input file.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/usage.rst:41]: Done

File: docs/source/usage.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.. _usage:
+
+Basic usage
+===========
+
+The main entry point is the ui.json (stored under ``assets``) for the octree-creation application that can be rendered
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/usage.rst:6]: Changed the layout all together. See what you think

File: docs/source/usage.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Alternatively, users can add the application to the choice list of ANALYST-Python scripts:
+
+.. figure:: ./images/dropdown.gif
+        :align: center
+        :width: 800
+
+Note that ANALYST needs to be restarted for the changes to take effect.
+
+Either operation will result in the rendering of a ui.json file within the Geoscience ANALYST viewport.
+To learn more about the ui.json interface visit `UIJson documentation <https://geoh5py.readthedocs.io/en/v0.8.0-rc.1/content/uijson_format/usage.html#usage-with-geoscience-analyst-pro>`_ page.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/usage.rst:31]: Agree, done

File: docs/source/usage.rst [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+===========
+
+The main entry point is the ui.json (stored under ``assets``) for the octree-creation application that can be rendered
+by Geoscience ANALYST. The interface has two sections described in the :ref:`Methodology <methodology>` section:
+
+ - :ref:`Mesh creation parameters <mesh_creation>`
+ - :ref:`Refinement parameters <refinements>`
+
+.. figure:: /images/ui_json.png
+    :scale: 40%
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/usage.rst:13]: Done - I think...

File: docs/source/methodology.rst [hunk 8]
```diff
@@ -0,0 +1,268 @@
+.. _methodology:
+
+Methodology
+===========
+
+This section provides technical details regarding the algorithm used for the
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:6]: Done

File: docs/source/methodology.rst [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+creation octree meshes. The entire process can be broken down into two main parts:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:7]: Done

File: docs/source/methodology.rst [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+
+- `Creating the base mesh (General Parameters) <mesh_creation>`_ that defines the outer extents and the core cell parameters.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:9]: Done

File: docs/source/methodology.rst [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+outer limits of the mesh.
+
+
+.. figure:: /images/extent_parameters.png
+    :width: 800
+
+- **Core hull extent**:
+
+    List of objects available in the target ``geoh5`` project. The base mesh will be centered on the
+    selected object and extend beyond its hull.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:29]: Base mesh includes padding and all, so kind of ambiguous. I can re-word.

File: docs/source/methodology.rst [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+- **Minimum Refinement**:
+
+    Largest octree level allowed after refinement.
+    The equivalent cell dimension is given by:
+
+    .. math::
+
+        h \times 2^{level - 1}
+
+    where *h* is the *core cell size* in a given direction.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:62]: True

[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:62]: Added a figure on top with more words

File: docs/source/methodology.rst [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+The final mesh expected would be a 512 x 512 x 128 cells mesh, with an extent of 12,800 x 12,800 x 3,200 m. Note that the
+cell size is uniform and equal to the minimum level of 5 (400 m cell size), as defined in the parameters.
+
+
+.. _refinements:
+
+Refinements
+-----------
+
+Once the extent of the mesh has been defined, the user can increase the resolution (add small cells) in specific regions of the mesh
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:143]: Yip

File: docs/source/methodology.rst (hunk 1 above)
Comment (domfournier) [reply docs/source/methodology.rst:167]: I changed my mind on naming - fixing.

File: docs/source/methodology.rst (hunk 3 above)
Comment (domfournier) [reply docs/source/methodology.rst:197]: Yip

File: docs/source/methodology.rst (hunk 2 above)
Comment (benk-mira) [reply docs/source/methodology.rst:189]: Doesn't the max distance apply to all the refinement types though?  So it should be at the same level as object/levels?

File: docs/source/methodology.rst [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+
+Example
+^^^^^^^
+
+The example below demonstrates this process with simple line survey shown below and the following parameters:
+
+.. image:: images/octree_padding_distance.png
+    :width: 800
+    :alt: paddings
+
```
Comment (benk-mira) [inline docs/source/methodology.rst:84]: This figure should have a caption making clear that the top image is plan view and bottom is section viewed from south (axes are hard to see)

**Review by benk-mira — CHANGES_REQUESTED** [review]
A couple more comments

File: docs/source/methodology.rst [hunk 15]
```diff
@@ -0,0 +1,286 @@
...
+outer limits of the mesh.
+
+
+.. figure:: /images/extent_parameters.png
+    :width: 800
+
+- **Core hull extent**:
+
+    List of objects available in the target ``geoh5`` project. The base mesh will be centered on the
+    selected object and fully containing its hull. If the mesh is to be used towards a forward simulation or an inversion,
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/methodology.rst:40]: nice

File: docs/source/methodology.rst (hunk 14 above)
Comment (domfournier) [reply docs/source/methodology.rst:84]: Roger

**Review by benk-mira — APPROVED** [review]

**Pull request merged**